File: PCH/DP/BOJ_9465.py
import sys
input = sys.stdin.readline

t = int(input())

for _ in range(t):
  n = int(input())
  
  dp = [[0]*100000,[0]*100000]
  
  for idx, var in enumerate(map(int,input().split())):
    dp[0][idx] = var
  for idx, var in enumerate(map(int,input().split())):
    dp[1][idx] = var
  
  dp[0][1] += dp[1][0]
  dp[1][1] += dp[0][0]
  
  # 맨 끝에서 시작 -> 둘 중 하나는 무조건 고르게 됨
  # ex - (1,위) 선택 => (0,아래) 선택됨
  # 2번 인덱스 부터는 선택지가 갈린다. 예를 들어 (2,위) 선택 시
  # (1) (1,아래)을 선택하면 (0,위)도 자동으로 선택된다.
  # (2) (0,아래) 선택
  # (1)과 (2) 중 큰 값을 선택한다.
  
  for i in range(2, n):
    dp[0][i] += max(dp[1][i-1], dp[1][i-2])
    dp[1][i] += max(dp[0][i-1], dp[0][i-2])
  
  print(max(dp[0][n-1], dp[1][n-1]))

File: LJH/DP/BOJ9465.js
// O(NlogN) 으로 풀어야 함
const fs = require("fs");
const filePath = process.platform === "linux" ? "/dev/stdin" : "./LJH/input.txt";
let input = fs.readFileSync(filePath).toString().trim().split("\n");
let t = Number(input.shift());
while (t--) {
    let col = Number(input.shift());
    let tRow = input.shift().split(" ").map(Number);
    let bRow = input.shift().split(" ").map(Number);
    let dp = [[0, tRow[0], bRow[0]]]; // [이전 열의 최대값, (이전 열 아래 행에 있는 값 + 현재 열 윗 행에 있는 값), (이전 열 윗 행에 있는 값 + 현재 열 아래 행에 있는 값)]
    for (let i = 1; i <= col; i++) {
        dp[i] = [
            Math.max(...dp[i - 1]), // 이전 열에서 아무것도 수행하지 않은 경우, 현재 열에서 위/아래 아무곳이나 선택 가능하므로 현재 열에서 값이 큰 행을 선택
            Math.max(dp[i - 1][0], dp[i - 1][2]) + tRow[i], // max(이전 열에서 아무것도 수행하지 않은 경우, 이전 열에서 아래 행을 골랐을 경우) + 현재 열의 위의 행
            Math.max(dp[i - 1][0], dp[i - 1][1]) + bRow[i], // max(이전 열에서 아무것도 수행하지 않은 경우, 이전 열에서 위의 행을 골랐을 경우) + 현재 열의 아래 행
        ];
    }
    console.log(Math.max(...dp[col - 1]));
}
